backend/src/controllers/admin_form.js

```diff
@@ -0,0 +1,78 @@
+const log = require("../facilities/logging.js")(module.filename);
```

Comment:
This might be used for all controllers, to optimize some of the code, and would rename it to something that makes more sense semantically